kedro/framework/project/__init__.py

```diff
@@ -259,3 +261,58 @@ def validate_settings():
         )
 
     importlib.import_module(f"{PACKAGE_NAME}.settings")
+
+
+def find_pipelines() -> Dict[str, Pipeline]:
+    """Automatically find modular pipelines having a ``create_pipeline``
+    function. By default, projects created using Kedro 0.18.3 and higher
+    call this function to autoregister pipelines upon creation/addition.
+
+    Returns:
+        A generated mapping from pipeline names to ``Pipeline`` objects.
+
+    Warns:
+        UserWarning: When a module does not expose a ``create_pipeline``
+            function, the ``create_pipeline`` function does not return a
+            ``Pipeline`` object, or if the module import fails up front.
+    """
+    pipelines_dict = {"__default__": pipeline([])}
+    for pipeline_dir in importlib_resources.files(
+        f"{PACKAGE_NAME}.pipelines"
+    ).iterdir():
+        if not pipeline_dir.is_dir():
+            continue
+        pipeline_name = pipeline_dir.name
+        if pipeline_name == "__pycache__":
+            continue
+        try:
+            pipeline_module = importlib.import_module(
+                f"{PACKAGE_NAME}.pipelines.{pipeline_name}"
+            )
+        except:  # pylint: disable=bare-except  # noqa: E722
+            warnings.warn(
+                f"An error occurred while importing the "
+                f"'{PACKAGE_NAME}.pipelines.{pipeline_name}' module. "
+                f"Nothing defined therein will be returned by "
+                f"'find_pipelines'."
+            )
```

Comment:
Is it worth putting the exception message in here too to aid further debugging?

Reply:
e.g. I am expecting that a common problem will be that people have the correct pipeline structure but have not installed the dependencies required by the pipeline. The solution to this (`pip install ...`) might be more obvious if the `ImportError` exception were shown. I guess the same exception might be shown when you do `kedro run` anyway though?

Reply:
I've updated to print the traceback in the warning:

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/14007150/184536222-f71702b4-6a8d-4de9-8014-4ee60ff2468c.png">

Does that work?

Reply:
This looks clear enough to me 👍